midend/interpreter.cpp

```diff
@@ -310,33 +310,50 @@ void SymbolicStruct::dbprint(std::ostream& out) const {
 }
 
 SymbolicHeaderUnion::SymbolicHeaderUnion(const IR::Type_HeaderUnion* type,
-                               bool uninitialized,
-                               const SymbolicValueFactory* factory) :
-        SymbolicStruct(type, uninitialized, factory),
-        valid(new SymbolicBool(false)) {}
+                                         bool uninitialized,
+                                         const SymbolicValueFactory* factory) :
+        SymbolicStruct(type, uninitialized, factory) {}
+
+void SymbolicHeaderUnion::setValid(bool v, cstring field) {
+    if (field) {
+        fieldValue[field]->to<SymbolicHeader>()->setValid(v);
+        for (auto f : type->to<IR::Type_StructLike>()->fields) {
+            if (f->name.name != field)
+                fieldValue[f->name.name]->setAllUnknown();
```

Comment:
I think you need to call `setValid(false)` here.

Reply:
I fixed it